### PR TITLE
feat(provider/kubernetes): v2 admission controllers

### PR DESCRIPTION
**Pull request by lwander**

### CODE REVIEW

**Comment by lwander** [comment]
https://github.com/spinnaker/spinnaker/issues/3007

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesMutatingWebhookConfigurationHandler.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  @Override
+  public boolean versioned() {
+    return false;
+  }
+
+  @Override
+  public KubernetesSpinnakerKindMap.SpinnakerKind spinnakerKind() {
+    return KubernetesSpinnakerKindMap.SpinnakerKind.UNCLASSIFIED;
```
Comment (jtk54) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesMutatingWebhookConfigurationHandler.java]: Didn't we declare a `Kind` for mutating webhooks?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesValidatingWebhookConfigurationHandler.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  @Override
+  public boolean versioned() {
+    return false;
+  }
+
+  @Override
+  public KubernetesSpinnakerKindMap.SpinnakerKind spinnakerKind() {
+    return KubernetesSpinnakerKindMap.SpinnakerKind.UNCLASSIFIED;
```
Comment (jtk54) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesValidatingWebhookConfigurationHandler.java]: Here too. I'm likely unfamiliar with the pattern but this seems confusing.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesMutatingWebhookConfigurationHandler.java (hunk 1 above)
Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesMutatingWebhookConfigurationHandler.java]: This is the spinnaker kind it maps to, not the Kubernetes kind. On startup, this spinnaker -> kubernetes kind relation is recorded, and informs both the resource providers (e.g. cluster provider, app provider, instance provider) how to serve these resources, as well as the UI on how to render them.

**Pull request merged**